core/templates/components/code-mirror/codemirror-mergeview.component.ts

```diff
@@ -27,14 +28,24 @@ import { downgradeComponent } from '@angular/upgrade/static';
 export class CodemirrorMergeviewComponent implements
   AfterViewInit, OnInit, OnChanges {
   @Input() options = {};
-  @Input() leftValue;
-  @Input() rightValue;
-  codeMirrorInstance: CodeMirror.MergeView.MergeViewEditor;
-  constructor(private elementRef: ElementRef, private ngZone: NgZone) { }
+  // These properties comes from 'state-diff-modal.controller.ts' file and can
+  // be undefined sometimes, after the view initializes.
+  @Input() leftValue: string = ' ';
+  @Input() rightValue: string = ' ';
+  // 'ngOnChanges' sometimes runs before the view has been initialized,
+  // to cater this we are checking it be to not undefined.
```

Comment:
Can you elaborate more on this? I don't really understand the comment.

Reply:
Now updated as per @vojtechjelinek suggested https://github.com/oppia/oppia/pull/13057#discussion_r650559166